src/mgr/Forwarder.cc

```diff
@@ -28,7 +28,10 @@ CBDATA_NAMESPACED_CLASS_INIT(Mgr, Forwarder);
 
 Mgr::Forwarder::Forwarder(const Comm::ConnectionPointer &aConn, const ActionParams &aParams,
                           HttpRequest* aRequest, StoreEntry* anEntry, const AccessLogEntryPointer &anAle):
-    Ipc::Forwarder(new Request(KidIdentifier, 0, aConn, aParams), 10),
+    // TODO: Add virtual Forwarder::makeRequest() to avoid prematurely creating
+    // this dummy request with a dummy ID that are finalized by Ipc::Forwarder.
+    // Same for Snmp::Forwarder.
+    Ipc::Forwarder(new Request(KidIdentifier, Ipc::RequestId(/*XXX*/), aConn, aParams), 10),
```

Comment:
Just to clarify: This TODO documents and offers a blueprint for resolving the XXX below it. The official code contains the same problem. Fixing this XXX (i.e. implementing that TODO) is far from trivial and will not cover any new use cases within this PR scope.  It should not be done in this (already rather large/complicated) PR.